Mu2eUtilities/inc/HelixTool.hh

```diff
@@ -60,10 +60,42 @@ namespace mu2e {
 
       if (counter > 0) _meanHitRadialDist /= counter;
 
-      _nLoops = (z_last_hit - z_first_hit)/(fabs(Helix->helix().lambda())*2.*M_PI);
+      _nLoops = (z_last_hit - z_first_hit)/(fabs(robustHel->lambda())*2.*M_PI);
 
       //here we evaluate once the impact parameter
       _d0 = robustHel->rcent  () - robustHel->radius ();
+
+
+      // we now estiamte the ratio of the number of measured hits to the number of the expected ones
+      // we make a few assumptions and appriximations:
+      //   - 36 tacking faces
+      //   - an average spacing of 178 mm
+      //   - we approximate the tracker as a cylinder: r_min = 380 mm, r_max = 860 mm
+      int      nFaces(36); 
+      float    expected_hits(0);
+      float    z_step(178.); 
+      float    minR_tracker(380.);
+      float    maxR_tracker(860.);
+
+      for (int i=0; i<nFaces;i++){
+	float   z = z_first_hit + (double)i*z_step;
+	if (z < z_first_hit )  continue;
+	if (z > z_last_hit  )  continue;
+
+	XYZVec  pos;
+	pos.SetZ(z);
+	robustHel->position(pos);
+      
+	//now check that we are in the active area of tracker
+	double   hitR = sqrt(pos.x()*pos.x() + pos.y()*pos.y());
+	if (hitR < minR_tracker)  continue;
+	if (hitR > maxR_tracker)  continue;
+
+	++expected_hits;
+      }
+
+      //each face has two planes. We are assuming 2 ComboHits per face
+      _hitRatio = nhits / (2.*expected_hits);
```

Comment:
This code block seems to be a copy of what's in the helix filter, please consolidate and make the changes suggested there.

Reply:
DONE